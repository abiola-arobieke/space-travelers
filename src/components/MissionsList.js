import React from 'react';
import PropTypes from 'prop-types';
import MissionItem from './MissionItem';
import '../styles/Missions.css';

const MissionsList = ({ missions }) => (
  <table className="missions-table">
    <thead>
      <tr>
        <th className="mission-th column-1">Mission</th>
        <th className="mission-th column-2">Description</th>
        <th className="mission-th column-3">Status</th>
        <th className="mission-th column-4"> </th>
      </tr>
    </thead>
    <tbody>
      {missions.data?.map((mission) => (
        <MissionItem key={mission.mission_id} mission={mission} />
      ))}
    </tbody>
  </table>
);

MissionsList.propTypes = {
  missions: PropTypes.shape({
    data: PropTypes.arrayOf(
      PropTypes.shape({
        mission_id: PropTypes.string.isRequired,
        mission_name: PropTypes.string.isRequired,
        description: PropTypes.string.isRequired,
      }),
    ),
  }).isRequired,
};

export default MissionsList;
