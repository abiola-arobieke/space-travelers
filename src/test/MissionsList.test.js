import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MissionsList from '../components/MissionsList';

const mockStore = configureStore([]);

describe('MissionsList', () => {
  it('renders correctly', () => {
    const missions = {
      data: [
        {
          mission_id: '1',
          mission_name: 'Mission 1',
          description: 'Mission description 1',
        },
        {
          mission_id: '2',
          mission_name: 'Mission 2',
          description: 'Mission description 2',
        },
      ],
    };

    const store = mockStore({ missions });

    const component = renderer.create(
      <Provider store={store}>
        <MissionsList missions={missions} />
      </Provider>,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
