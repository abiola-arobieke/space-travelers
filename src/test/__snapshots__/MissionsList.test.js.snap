// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsList renders correctly 1`] = `
<table
  className="missions-table"
>
  <thead>
    <tr>
      <th
        className="mission-th column-1"
      >
        Mission
      </th>
      <th
        className="mission-th column-2"
      >
        Description
      </th>
      <th
        className="mission-th column-3"
      >
        Status
      </th>
      <th
        className="mission-th column-4"
      >
         
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="row-cell"
    >
      <td
        className="mission-td mission-name"
      >
        Mission 1
      </td>
      <td
        className="mission-td mission-description"
      >
        Mission description 1
      </td>
      <td
        className="mission-td mission-member"
      >
        <button
          className="btn btn-secondary member-button"
          type="button"
        >
          NOT A MEMBER
        </button>
      </td>
      <td
        className="mission-td mission-join"
      >
        <button
          className="btn btn-outline-secondary join-button"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr
      className="row-cell"
    >
      <td
        className="mission-td mission-name"
      >
        Mission 2
      </td>
      <td
        className="mission-td mission-description"
      >
        Mission description 2
      </td>
      <td
        className="mission-td mission-member"
      >
        <button
          className="btn btn-secondary member-button"
          type="button"
        >
          NOT A MEMBER
        </button>
      </td>
      <td
        className="mission-td mission-join"
      >
        <button
          className="btn btn-outline-secondary join-button"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
